static/src/javascripts/projects/common/modules/commercial/braze/brazeBanner.tsx

```diff
@@ -106,13 +125,16 @@ const SDK_OPTIONS = {
 	devicePropertyAllowlist: [],
 };
 
-const getMessageFromBraze = async (apiKey, brazeUuid) => {
+const getMessageFromBraze = async (
+	apiKey: string,
+	brazeUuid: string,
+): Promise<BrazeMessage> => {
 	const sdkLoadTiming = measureTiming('braze-sdk-load');
 	sdkLoadTiming.start();
 
-	const appboy = await import(
+	const importedAppboy = (await import(
 		/* webpackChunkName: "braze-web-sdk-core" */ '@braze/web-sdk-core'
-	);
+	)) as unknown as typeof appboy;
```

Comment:
Is this because the package does not have type declarations?

Another approach is:
```
// @ts-expect-error -- no types are exported from this lib (or any message you like)
const importedAppboy: typeof appboy = (await import(
		/* webpackChunkName: "braze-web-sdk-core" */ '@braze/web-sdk-core'
	));
```

Reply:
> Is this because the package does not have type declarations?

No, the imported package does have type declarations. This was the only approach I found that worked to get those types to flow through. We use the same approach on DCR, but it was a while ago so I don't remember all the details. Possibly it was to do with the lazy/runtime import we're doing here and TS not being able to follow that? In the package `appboy` is a namespace which always felt slightly odd to me.

Reply:
I had a look [at DCR](https://github.com/guardian/dotcom-rendering/blob/59677d51dda690f0afbf8e1337de5d091be62df9/dotcom-rendering/src/web/lib/braze/initialiseAppboy.ts#L14-L16), and realised what the issue is: you should destructure the `default` export!

```ts
// Happy days
const { default: importedAppboy } = await import('@braze/web-sdk-core');
// const importedAppboy: typeof appboy
```

Reply:
Oh right! Yeah that checks out with the docs [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#importing_defaults). But the existing code definitely functions how we want. So are:

```js
const importedAppboy = await import(...)
```

and:

```js
const { default: importedAppboy } = await import(...)
```

functionally equivalent? I'm definitely going to make the change you suggested, but just a bit confused about the existing behaviour.

Reply:
Yes, my understanding is that they are functionally equivalent, [which the MDN docs seem to coroborate](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#return_value).